goclean.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+# The script does automatic checking on a Go package and its sub-packages, including:
+# 1. gofmt         (http://golang.org/cmd/gofmt/)
+# 2. goimports     (https://github.com/bradfitz/goimports)
+# 3. golint        (https://github.com/golang/lint)
+# 4. go vet        (http://golang.org/cmd/vet)
+# 5. race detector (http://blog.golang.org/race-detector)
+# 6. test coverage (http://blog.golang.org/cover)
+
+set -e
+
+# Automatic checks
+test -z $(gofmt -l -w . | tee /dev/stderr)
+test -z $(goimports -l -w . | tee /dev/stderr)
+test -z $(golint ./... | grep -v "ALL_CAPS\|OP_\|NewFieldVal\|Id\|RpcCommand\|RpcRawCommand\|RpcSend\|Dns" | tee /dev/stderr)
+go tool vet -structtags=false .
+env GORACE="halt_on_error=1" go test -v -race ./...
+
+# Run test coverage on each subdirectories and merge the coverage profile.
+
+echo "mode: count" > profile.cov
+
+# Standard go tooling behavior is to ignore dirs with leading underscores.
+for dir in $(find . -maxdepth 10 -not -path '.' -not -path './.git*' \
+    -not -path '*/_*' -not -path './cmd*' -not -path './release*' -type d)
```

Comment:
These too:

`for dir in $(find . -maxdepth 10 -not -path './.git*' -not -path '*/_*' -type d)`